tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,268 @@
+mod error;
+mod keys;
+mod root;
+mod s3;
+mod shared;
+
+use error::Result;
+use log::{error, info};
+use pubsys_config::InfraConfig;
+use sha2::{Digest, Sha512};
+use shared::KeyRole;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::path::{Path, PathBuf};
+use std::{fs, process};
+use structopt::StructOpt;
+use tokio::runtime::Runtime;
+use url::Url;
+
+//   =^..^=   =^..^=   =^..^=  SUB-COMMAND STRUCTS  =^..^=   =^..^=   =^..^=
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+struct Args {
+    #[structopt(global = true, long, default_value = "INFO")]
+    log_level: LevelFilter,
+
+    // Path to Infra.toml  (NOTE: must be specified before subcommand)
+    #[structopt(long, parse(from_os_str))]
+    infra_config_path: PathBuf,
+
+    #[structopt(subcommand)]
+    subcommand: SubCommand,
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+struct CreateInfraArgs {
+    #[structopt(long)]
+    root_role_path: PathBuf,
+}
+
+#[derive(Debug, StructOpt)]
+enum SubCommand {
+    CheckInfraLock,
+    CreateInfra(CreateInfraArgs),
+}
+
+//  =^..^=   =^..^=   =^..^=  MAIN METHODS  =^..^=   =^..^=   =^..^=
+
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    // Parse and store the args passed to the program
+    let args = Args::from_args();
+
+    SimpleLogger::init(args.log_level, LogConfig::default()).context(error::Logger)?;
+
+    match args.subcommand {
+        SubCommand::CheckInfraLock => {
+            let rt = Runtime::new().context(error::Runtime)?;
+            rt.block_on(async { check_infra_lock(&args.infra_config_path).await })
+        }
+        SubCommand::CreateInfra(ref run_task_args) => {
+            let rt = Runtime::new().context(error::Runtime)?;
+            rt.block_on(async {
+                create_infra(&args.infra_config_path, &run_task_args.root_role_path).await
+            })
+        }
+    }
+}
+
+async fn check_infra_lock(toml_path: &Path) -> Result<()> {
+    let lock_path = toml_path
+        .parent()
+        .context(error::Parent { path: toml_path })?
+        .join("Infra.lock");
+
+    ensure!(!lock_path.is_file(), {
+        error!(
+            "It looks like you've already created some resources for your custom TUF repository because a lock file exists at '{}'.
+            \nPlease clean up your TUF resources in AWS, delete Infra.lock, and run again.",
+            lock_path.display()
+        );
+        error::FileExists { path: lock_path }
+    });
+    Ok(())
+}
+
+/// Automates setting up infrastructure for a custom TUF repo
+async fn create_infra(toml_path: &Path, root_role_path: &Path) -> Result<()> {
+    info!("Parsing Infra.toml...");
```

Comment:
Although we have the subcommand for manually ensuring that Infra.lock does not exist, do you think we should add that check here to make sure we don't do bad things™ by accident?

```rust
// TODO - support updating existing infra
check_infra_lock(/*..*/)?;
```

Reply:
Agreed! I think we should remove that subcommand entirely and just have it as a method in the file. It seems we are checking Infra.lock existence via code in pubsys anyway. Maybe later if we need the subcommand, we can add it back (doesn't seem like we need it now, if I'm not mistaken).